File: 1927-sum-game/1927-sum-game.java
class Solution {
    public boolean sumGame(String num) {
        int n = num.length() / 2;
        int sLeft = 0, qLeft = 0;
        int sRight = 0, qRight = 0;

        // Calculate sum and question mark count for left half
        for (int i = 0; i < n; i++) {
            char c = num.charAt(i);
            if (c == '?') {
                qLeft++;
            } else {
                sLeft += c - '0';
            }
        }

        // Calculate sum and question mark count for right half
        for (int i = n; i < num.length(); i++) {
            char c = num.charAt(i);
            if (c == '?') {
                qRight++;
            } else {
                sRight += c - '0';
            }
        }

        int sDiff = sLeft - sRight;
        int qDiff = qLeft - qRight;

        // Bob wins if and only if sDiff + 4.5 * qDiff == 0
        return (sDiff * 2 + qDiff * 9) != 0;
    }
}

File: 1927-sum-game/1927-sum-game.py
class Solution:
    def sumGame(self, num: str) -> bool:
        n = len(num) // 2
        
        # Calculate sum and count of '?' for left half
        s_left = sum(int(c) for c in num[:n] if c != '?')
        q_left = num[:n].count('?')
        
        # Calculate sum and count of '?' for right half
        s_right = sum(int(c) for c in num[n:] if c != '?')
        q_right = num[n:].count('?')
        
        # Total sum difference and question mark count difference
        s_diff = s_left - s_right
        q_diff = q_left - q_right
        
        # Bob wins if and only if s_diff + 4.5 * q_diff == 0
        return (s_diff * 2 + q_diff * 9) != 0